src/corehost/common/pal.unix.cpp

```diff
@@ -816,11 +806,12 @@ static void readdir(const pal::string_t& path, const pal::string_t& pattern, boo
                 continue;
             }
 
-            pal::string_t filepath(entry->d_name);
-            if (filepath != _X(".") && filepath != _X(".."))
+            if (!strcmp(entry->d_name, _X(".")) || !strcmp(entry->d_name, _X("..")))
```

Comment:
A nit - we can remove the _X(...) wrappers here. We know that the strcmp needs plain 8 bit char strings

Reply:
Done. Force-pushed with the changes, and rebased to current HEAD.